src/helper-modules/ordnance-survey/ordnance-survey.service.test.ts

```diff
@@ -0,0 +1,126 @@
+
+import { OrdnanceSurveyException } from './exception/ordnance-survey.exception';
+import { OrdnanceSurveyService } from './ordnance-survey.service';
+
+describe('OrdnanceSurveyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+
+  let httpServiceGet: jest.Mock;
+  let configServiceGet: jest.Mock;
+  let service: OrdnanceSurveyService;
+
+  const testPostcode = GEOSPATIAL.EXAMPLES.POSTCODE;
+  const testKey = valueGenerator.string({ length: 10 });
+  const basePath = '/search/places/v1/postcode';
+
+  beforeEach(() => {
+    const httpService = new HttpService();
+    const configService = new ConfigService();
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    configServiceGet = jest.fn().mockReturnValue({ key: testKey });
+    configService.get = configServiceGet;
+
+    service = new OrdnanceSurveyService(httpService, configService);
+  });
+
+  describe('getAddressesByPostcode', () => {
+    const expectedPath = `${basePath}?postcode=${encodeURIComponent(testPostcode)}&lr=EN&key=${encodeURIComponent(testKey)}`;
+
+    const expectedHttpServiceGetArgs: [string, object] = [expectedPath, { headers: { 'Content-Type': 'application/json' } }];
+
+    it('sends a GET to the Ordnance Survey API /search endpoint with the specified request', async () => {
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(
+          of({
+            data: expectedResponse,
+            status: 200,
+            statusText: 'OK',
+            config: undefined,
+            headers: undefined,
+          }),
+        );
+
+      await service.getAddressesByPostcode(testPostcode);
+
+      expect(httpServiceGet).toHaveBeenCalledTimes(1);
+      expect(httpServiceGet).toHaveBeenCalledWith(...expectedHttpServiceGetArgs);
+    });
+
+    it.each([
+      {
+        postcode: 'W1A 1AA',
+        expectedUrlQueryPart: `?postcode=W1A%201AA`,
+      },
+      {
+        postcode: 'W1A1AA',
+        expectedUrlQueryPart: '?postcode=W1A1AA',
+      },
+    ])('call Ordnance Survey API with correct and safe query parameters "$expectedUrlQueryPart"', async ({ postcode, expectedUrlQueryPart }) => {
+      const expectedPath = `${basePath}${expectedUrlQueryPart}&lr=EN&key=${encodeURIComponent(testKey)}`;
+      const expectedHttpServiceGetArgs: [string, object] = [expectedPath, { headers: { 'Content-Type': 'application/json' } }];
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(
+          of({
+            data: expectedResponse,
+            status: 200,
+            statusText: 'OK',
+            config: undefined,
+            headers: undefined,
+          }),
+        );
+
+      await service.getAddressesByPostcode(postcode);
+
+      expect(httpServiceGet).toHaveBeenCalledTimes(1);
+      expect(httpServiceGet).toHaveBeenCalledWith(...expectedHttpServiceGetArgs);
+    });
```

Comment:
As far as I can tell, the `it('sends a GET to the Ordnance Survey API /search endpoint with the specified request'...` block and the `it.each` block below it test exactly the same thing, but have different names. Could we combine these into a single `describe.each` block which tests each of the three postcodes, with two `it` blocks inside:
1. This existing test that we call Ordnance Survey with the correct arguments
2. A new test to check that we return the `expectedResponse`